libursa/src/signatures/ed25519.rs

```diff
@@ -22,6 +22,9 @@ impl Ed25519Sha512 {
     pub fn ver_key_to_key_exchange(pk: &PublicKey) -> Result<PublicKey, CryptoError> {
         use curve25519_dalek::edwards::CompressedEdwardsY;
 
+        if pk.len() != PUBLIC_KEY_SIZE {
```

Comment:
In my opinion it rather `PublicKey` has to contain the method `new or from_bytes`() and do length checking there.
Else you have to put the same check in every function using PublicKey

Reply:
`PublicKey` is a generic class that can be variable in length depending on which Crypto Primitive is using it. So it must be done here.